<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ReplaceDialog</class>
 <widget class="QDialog" name="ReplaceDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>542</width>
    <height>330</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Find or Replace a text</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout">
     <property name="margin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>6</number>
     </property>
     <item row="1" column="0">
      <widget class="QCheckBox" name="m_replaceCheckBox">
       <property name="whatsThis">
        <string>p&gt;If checked, &lt;b&gt;XINX&lt;/b&gt; replace the searched string by this  sting.&lt;/p&gt;
&lt;p&gt;If the search string is a &lt;i&gt;regular expression&lt;/i&gt;, you can use \1 or \2 for captured string. Captured string in the find string is in bracket.&lt;/p&gt;</string>
       </property>
       <property name="text">
        <string>&amp;Replace by :</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="m_comboFind">
       <property name="whatsThis">
        <string>&lt;p&gt;Text to search in the editor. The search string can be &lt;i&gt;string literal&lt;/i&gt; or a &lt;i&gt;regular expression&lt;/i&gt; (if the corresponding checkbox is checked).&lt;/p&gt;
&lt;p&gt;&lt;b&gt;XINX&lt;/b&gt; keep history of searched text.&lt;/p&gt;</string>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="currentIndex">
        <number>-1</number>
       </property>
       <property name="insertPolicy">
        <enum>QComboBox::InsertAlphabetically</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Find &amp;what :</string>
       </property>
       <property name="buddy">
        <cstring>m_comboFind</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="m_comboReplace">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="insertPolicy">
        <enum>QComboBox::InsertAlphabetically</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_2">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>&amp;Where</string>
       </property>
       <property name="buddy">
        <cstring>m_directoryWidget</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="DirectoryEditWidget" name="m_directoryWidget" native="true">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
       <property name="directory" stdset="0">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout">
     <property name="margin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>6</number>
     </property>
     <item row="1" column="1">
      <spacer>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="1">
      <widget class="QGroupBox" name="m_directionGroupBox">
       <property name="whatsThis">
        <string>Direction where &lt;b&gt;XINX&lt;/b&gt; must go for search string.</string>
       </property>
       <property name="title">
        <string>Direction</string>
       </property>
       <layout class="QVBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="margin">
         <number>9</number>
        </property>
        <item>
         <widget class="QRadioButton" name="m_UpRadioButton">
          <property name="text">
           <string>Go up</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="m_DownRadioButton">
          <property name="text">
           <string>Go down</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="2">
      <spacer>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="2">
      <widget class="QGroupBox" name="m_selectionGroupBox">
       <property name="whatsThis">
        <string>If &lt;i&gt;All&lt;/i&gt; is checked &lt;b&gt;XINX&lt;/b&gt; search in the whole document, else &lt;b&gt;XINX&lt;/b&gt; search only in the selection.</string>
       </property>
       <property name="title">
        <string>Extend</string>
       </property>
       <layout class="QVBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="margin">
         <number>9</number>
        </property>
        <item>
         <widget class="QRadioButton" name="m_searchAllRadioButton">
          <property name="text">
           <string>Current document</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="m_searchSelectionRadioButton">
          <property name="text">
           <string>Selection</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="m_projectFilesRadioButton">
          <property name="text">
           <string>Projects files</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="m_customFilesRadioButton">
          <property name="text">
           <string>Custom files</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="0" rowspan="2">
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Options</string>
       </property>
       <layout class="QVBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="margin">
         <number>9</number>
        </property>
        <item>
         <widget class="QCheckBox" name="m_caseCheckBox">
          <property name="whatsThis">
           <string>If checked, &lt;b&gt;XINX&lt;/b&gt; search the case sensitive string.</string>
          </property>
          <property name="text">
           <string>Match c&amp;ase</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="m_fromStartCheckBox">
          <property name="whatsThis">
           <string>If checked, &lt;b&gt;XINX&lt;/b&gt; search the word from the start of the document, else &lt;b&gt;XINX&lt;/b&gt; search from the cursor position.</string>
          </property>
          <property name="text">
           <string>Search from &amp;start</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="m_wholeWordsCheckBox">
          <property name="whatsThis">
           <string>If unchecked, &lt;b&gt;XINX&lt;/b&gt; search string include in other word, else only whole word is searched.</string>
          </property>
          <property name="text">
           <string>&amp;Whole words</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="m_regularExpressionCheckBox">
          <property name="whatsThis">
           <string>If checked, &lt;b&gt;XINX&lt;/b&gt; use regular expression to match string.</string>
          </property>
          <property name="text">
           <string>&amp;Use regular expression</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="m_buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DirectoryEditWidget</class>
   <extends>QWidget</extends>
   <header location="global">directoryedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_comboFind</tabstop>
  <tabstop>m_replaceCheckBox</tabstop>
  <tabstop>m_comboReplace</tabstop>
  <tabstop>m_directoryWidget</tabstop>
  <tabstop>m_caseCheckBox</tabstop>
  <tabstop>m_fromStartCheckBox</tabstop>
  <tabstop>m_wholeWordsCheckBox</tabstop>
  <tabstop>m_regularExpressionCheckBox</tabstop>
  <tabstop>m_UpRadioButton</tabstop>
  <tabstop>m_DownRadioButton</tabstop>
  <tabstop>m_searchAllRadioButton</tabstop>
  <tabstop>m_searchSelectionRadioButton</tabstop>
  <tabstop>m_projectFilesRadioButton</tabstop>
  <tabstop>m_customFilesRadioButton</tabstop>
 </tabstops>
 <resources>
  <include location="../../xinx/application.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_replaceCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_comboReplace</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_DownRadioButton</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_searchAllRadioButton</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_UpRadioButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_searchSelectionRadioButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_regularExpressionCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_wholeWordsCheckBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_regularExpressionCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_caseCheckBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_customFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_directoryWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_customFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_2</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>87</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_projectFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_directionGroupBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_projectFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_fromStartCheckBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_customFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_directionGroupBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>239</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_customFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_fromStartCheckBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_customFilesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_2</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>450</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_projectFilesRadioButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>191</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_fromStartCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_customFilesRadioButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ReplaceDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ReplaceDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>540</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
